### PR TITLE
Improve memory info support

### LINKED ISSUE #777: Hypre, Umpire and hipMemGetInfo
Hi All,
I'm using Hypre with Umpire enabled for AMD GPUs (rocm 5.1.0). In my application, I query the amount of free memory available with hipMemGetInfo after key steps (matrix creation, vector creation, preconditioner setup, ...). The results of these calls when not using Umpire are:

Before BuildMatrix : rank 0 of 8, free memory=63.98 GB, total memory=63.98 GB
Before hypre_CSRMatrixGpuRawSpMVAnalysis : rank 0 of 8, free memory=55.32 GB, total memory=63.98 GB
Before RHS IJVectorCreate : rank 0 of 8, free memory=54.90 GB, total memory=63.98 GB
After Preconditioner Setup : rank 0 of 8, free memory=43.68 GB, total memory=63.98 GB

This seems reasonable to me given the size of the problem. Once I build with Hypre with Umpire and a 4GB Device Memory Pool, I see:

Before BuildMatrix : rank 0 of 8, free memory=63.98 GB, total memory=63.98 GB
Before hypre_CSRMatrixGpuRawSpMVAnalysis : rank 0 of 8, free memory=.8517 GB, total memory=63.98 GB
Before RHS IJVectorCreate : rank 0 of 8, free memory=.8517 GB, total memory=63.98 GB
After Preconditioner Setup : rank 0 of 8, free memory=17.73 GB, total memory=63.98 GB

My question is, should I expect hipMemGetInfo to work properly when using device memory pools?
Thanks
-Paul

### PR DESCRIPTION
**Pull request by victorapm**
This PR includes changes to memory usage reporting and GPU support in hypre. Changes are summarized below:

-  Retrieve overall GPU memory usage (broader than umpire's pool info)
-  Update memory usage reporting to use gibibytes (GiB) instead of gigabytes (GB)
-  Enhance the memory printing functionality.

Closes #777

### CODE REVIEW

**Comment by rfalgout** [comment]
Quick question @victorapm .  Why is GiB preferred over GB?  They both have a specific definition, though GB is sometimes incorrectly used to mean GiB, while the latter is probably less recognizable.  Just wondering what your thought process was here.  Thanks!

**Comment by victorapm** [comment]
Hi Rob, I chose to report GiB because that's easier to relate to the specs of a card on paper, even though marketing uses GB and GiB interchangeably. For example, the MI250Xs have 128 GiB of HBM2e memory

**Comment by rfalgout** [comment]
> Hi Rob, I chose to report GiB because that's easier to relate to the specs of a card on paper, even though marketing uses GB and GiB interchangeably. For example, the MI250Xs have 128 GiB of HBM2e memory

Sounds good.  Thanks!

File: src/utilities/device_utils.c [hunk 1]
```diff
@@ -348,6 +348,51 @@ hypre_ForceSyncComputeStream()
 
 #if defined(HYPRE_USING_GPU)
 
+/*--------------------------------------------------------------------------
+ * hypre_DeviceMemoryGetUsage
+ *
+ * Retrieves memory usage statistics for the current GPU device. The function
+ * fills an array with memory data, converted to gigabytes (GB):
```
Comment (rfalgout) [inline src/utilities/device_utils.c]: GiB

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

File: src/utilities/device_utils.c (hunk 1 above)
Comment (victorapm) [reply src/utilities/device_utils.c]: Thanks for catching!

**Pull request merged**